### PR TITLE
Prioritise testing based on contacts

### PR DESCRIPTION
**Pull request by danielmonterocr**
@roberthinch please take a look, as of now every age group can take a priority test.

### CODE REVIEW

**Comment by danielmonterocr** [comment]
@roberthinch I created 3 different configurations and they run randomly on every PR, the other option is to create 3 different tests for each. Please take a look.

File: tests/test_interventions.py [hunk 1]
```diff
@@ -1203,7 +1236,148 @@ def test_quarantined_have_trace_token(self, test_params, time_steps_test ):
...
+        pre_df_symp  = pd.merge( pre_df_symp, pre_df_trace_sum, left_on = "ID_recipient", right_on = "index_ID", how = "left" )
+        pre_df_symp["priority"] = ( pre_df_symp["n_interactions"] > priority_test_contacts )
+
+        # check we have sufficient priority and non-priority to test
+        if ( ( params.get_param( "test_order_wait_priority" ) != -1 ) and ( params.get_param( "test_result_wait_priority" ) != -1 ) ):
+            np.testing.assert_equal(sum(pre_df_symp["priority"]== True) > 30, True, "Not sufficient priority cases to meaningfully test" )
+        np.testing.assert_equal(sum(pre_df_symp["priority"]== False) > 30, True, "Not sufficient non-priority cases to meaningfully test" )
+        
+        # check the test results being received are in line with whether it is a priority test
+        if ( params.get_param( "test_order_wait_priority" ) != -1 and params.get_param( "test_result_wait_priority" ) != -1 ):
```
Comment (roberthinch) [inline tests/test_interventions.py:1322]: This currently only works if `test_order_wait_priority == 0`. So, you need to look at people who become symptomatic `test_order_wait_priority` steps prior to `test_time` at line 1305. I expect it is a one line change. One of your tests parameter sets should have `test_order_wait=2` and `test_order_wait_priority = 1`.

File: tests/test_interventions.py [hunk 2]
```diff
@@ -1203,7 +1236,148 @@ def test_quarantined_have_trace_token(self, test_params, time_steps_test ):
...
+        non_priority_traced = df_trace_sum[ ( df_trace_sum["priority"]==False) & ( df_trace_sum["n_quar_uniq"] > 1 )]
+        np.testing.assert_equal( sum( df_trace_sum["priority"]==False ) > 30, True, "Insufficient non-priority cases to test" )
+        np.testing.assert_equal( len( non_priority_traced ), 0, "Non-priority index cases traced people quarantined too soon" )
+        
+        # do a dirty measure rough measure as well to make sure not all are multiply traced
+        non_priority_traced2 = df_trace_sum[ ( df_trace_sum["priority"]==False ) & ( df_trace_sum["n_interactions"] == df_trace_sum["n_quar"] ) ]
+        np.testing.assert_equal( len( non_priority_traced2 ), 0, "Non-priority index cases traced people quarantined too soon (rough measure)" )
+
+        # next step forward to when the non-priority test results are expected
+        for time in range( params.get_param( "test_order_wait" ) + params.get_param( "test_result_wait" ) - params.get_param( "test_result_wait_priority" )):
```
Comment (roberthinch) [inline tests/test_interventions.py:1370]: I think this should have a  `-params.get_param( "test_order_wait_priorty")` as well. Without I expect it fails the test case `test_order_wait=2` and `test_order_wait_priority = 1`.

File: tests/test_interventions.py [hunk 3]
```diff
@@ -1203,7 +1236,148 @@ def test_quarantined_have_trace_token(self, test_params, time_steps_test ):
...
+        elif ( test_priority_config == 3):
+            params.set_param( "test_order_wait", 3 )
+            params.set_param( "test_result_wait", 2 )
+            params.set_param( "test_order_wait_priority", -1 )
+            params.set_param( "test_result_wait_priority", -1 )
+        params = utils.set_app_users_fraction_all( params, app_users_fraction)
+        params = utils.set_priority_test_contacts_all( params, priority_test_contacts )  
+        model  = utils.get_model_swig( params )
+
+        print(f"test_time {test_time}")
```
[the remark this answers is not in the document]
Comment (danielmonterocr) [reply tests/test_interventions.py:1298]: They are just for debugging, I'll get rid of them before merging

File: src/interventions.c [hunk 4]
```diff
@@ -436,8 +483,17 @@ void intervention_test_order( model *model, individual *indiv, int time )
 {
 	if( indiv->quarantine_test_result == NO_TEST && !(indiv->infection_events->is_case) )
 	{
-		add_individual_to_event_list( model, TEST_TAKE, indiv, time );
-		indiv->quarantine_test_result = TEST_ORDERED;
+        int traceable_inter = number_of_traceable_interactions( model, indiv );
```
[the remark this answers is not in the document]
Comment (danielmonterocr) [reply src/interventions.c]: Where should I set this default value? in the baseline params file?

File: tests/test_interventions.py (hunk 1 above)
Comment (danielmonterocr) [reply tests/test_interventions.py:1368]: @roberthinch made this change and now I'm getting a seg fault, could you please take a look at the test? I uploaded the latest version

**Pull request merged**